packages/safe-tools-client/app/services/network.ts

```diff
@@ -50,7 +57,12 @@ export default class NetworkService extends Service {
       }))
       .filter(({ name }) => name !== this.name);
   }
-  @action onSelect(networkInfo: NetworkInfo) {
+
+  @action async onSelect(networkInfo: NetworkInfo) {
+    await this.wallet.switchNetwork(networkInfo.chainId);
+
+    if (this.networkInfo.chainId === networkInfo.chainId) return;
```

Comment:
When switching the network, this action will be called twice - firstly from the dropdown and secondly when metamask changes network. In this case, we don't want to assign the same network twice. If we assign the same thing, safe info area will flicker because it will try to refetch token balances.